### PR TITLE
Add support for spring type conversion

### PR DESCRIPTION
**Pull request by sebastian-mikucki**
@tbradlo @dominik1102 @rmycek @Konrad-Rys 
I've added support for spring type conversion, which also adds support for converting string to enum by default, so in controller you can type enum type instead of String and then valueOf(string), I've also refactor CarConverter to implement Converter and convert cars via ConversionService

### CODE REVIEW

**Comment by sebastian-mikucki** [comment]
Change GameConverter, MapConverter, MovementHistoryConverter

**Comment by tbradlo** [comment]
enums are nice. Can we inject Entities instead of DTO (if enums we can)? How is it different from: http://www.baeldung.com/entity-to-and-from-dto-for-a-java-spring-application ?

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 1]
```diff
@@ -30,28 +30,31 @@
...
     @ApiResponses(value = {
             @ApiResponse(code = 201, message = "Car successfully added")
     })
-    public Car addCar(@PathVariable("name") String name, @RequestBody String type){
-        CarType carType = CarType.valueOf(type);
-
+    public Car addCar(@PathVariable("name") String name, @RequestBody CarType carType){
         CarEntity addedCar = carService.addCar(carType, name);
-        return toCar(addedCar);
+        return conversionService.convert(addedCar, Car.class);
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: what about conversionService.toCar() ?

File: src/main/java/com/guysfromusa/carsgame/v1/CarResource.java [hunk 2]
```diff
@@ -30,28 +30,31 @@
...
     }
 
     @GetMapping()
     @ApiOperation(value = "Find all available cars", response = List.class)
     public List<Car> getAllCars(){
         List<CarEntity> carEntities = this.carService.loadAllCars();
-        return toCars(carEntities);
+        return carEntities.stream()
+                .map(carEntity -> conversionService.convert(carEntity, Car.class))
+                .collect(Collectors.toList());
```
Comment (tbradlo) [inline src/main/java/com/guysfromusa/carsgame/v1/CarResource.java]: maybe come guysfromusa.StrimUtils.transformList(list, lambda) ?

**Pull request merged**